roofit/roofitcore/src/RooMinimizerFcn.cxx

```diff
@@ -70,21 +65,17 @@ RooMinimizerFcn::RooMinimizerFcn(RooAbsReal *funct, RooMinimizer* context,
   _constParamList->setName("constParamList");
 
   // Remove all non-RooRealVar parameters from list (MINUIT cannot handle them)
-  TIterator* pIter = _floatParamList->createIterator();
-  RooAbsArg* arg;
-  while ((arg=(RooAbsArg*)pIter->Next())) {
+  for (unsigned int i = 0; i < _floatParamList->size(); ++i) { // Note: Counting loop, since removing from collection!
+    const RooAbsArg* arg = (*_floatParamList).at(i);
```

Comment:
I'm not sure I understand: if `_floatParamList->remove(*arg)` removes element 3, do elements after that shift back one position? if yes, aren't you skipping the new 3/old 4 element when incrementing i after a removal?

Reply:
Probably yes. This could be a bug both in old and new loop!